### PR TITLE
[WIP] Initial stab at Nature blog post.

### PR DESCRIPTION
**Pull request by rgbkrk**
Here's the start of a blog post for the [nature demo](http://www.nature.com/news/ipython-interactive-demo-7.21492?article=1.16261).

Since I'm letting myself get stuck in writer's block, I'm pushing this out as a WIP and to ask for feedback from colleagues.

### CODE REVIEW

File: src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+### Static assets should be served via nginx instead of within each user container
+
+```
+location ~ /(user[-/][a-zA-Z0-9]*)/static/(.*) {
+    alias /srv/ipython/IPython/html/static/$2;
+}
+```
+
+This is both for speed and to use less file descriptors across the system.
```
Comment (smashwilson) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: "... to use fewer file descriptors..."

File: src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+We ended up pulling some neat tricks with our deployment setup to [mount a dummy user container's static files into an nginx container](https://github.com/jupyter/tmpnb-deploy/pull/3).
+
+HOWEVER, if you don't do this, you get the neat side effect of never having caching problems when launching userland containers and even means you could ship different ones to different users!
```
Comment (smashwilson) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: I wouldn't capitalize "HOWEVER" here. Also, you might consider combining these three paragraphs into one. Maybe (with another tweak or two):

> This is both for speed and to use fewer file descriptors across the system. We ended up pulling some neat tricks with our deployment setup to [mount a dummy user container's static files into an nginx container](https://github.com/jupyter/tmpnb-deploy/pull/3). However, if you don't do this, you get the neat side effect of never having caching problems when launching new versions of userland containers, and even means you could ship different ones to different users!

File: src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+< Note about MozFest, London, Nature offices >
+
+### Redirection
+
+
+
+## Closing up
+
+We love IPython notebooks, the overall architecture that has been built out here, and hope to keep supporting Open Source projects do interesting things on the internet in a way that benefits community, technology, and the whole ecosystem.
```
Comment (smashwilson) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: :+1: :rainbow:

File: src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+```
+https://tmpnb.org/user/FKlVK3haOQRF/notebooks/SharingEffects.ipynb
+```
+
+### Pool user containers ahead of time
+
+Docker can only boot and route so many containers so quickly. To mitigate this, we created a pool of ready to go userland containers. [@smashwilson](https://github.com/smashwilson) came in and added this
+
+[![pooling.gif](https://d23f6h5jpj26xu.cloudfront.net/jlvadowzumttlg_small.gif)](http://img.svbtle.com/jlvadowzumttlg.gif)
```
Comment (smashwilson) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: Oh good grief that giant goofy smile

Comment (rgbkrk) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: Drawn by le @jnoller.

File: src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+categories:
+    - IPython
+    - Science
+    - Data Science
+    - Docker
+    - tmpnb
+    - Jupyter
+    - Ephemeral
+---
+The IPython/Jupyter notebook is a wonderful environment for computations, prose, plots, and interactive widgets that you can share with collaborators. People use the notebook [all](http://blog.quantopian.com/quantopian-research-your-backtesting-data-meets-ipython-notebook/) [over](http://nbviewer.ipython.org/github/GoogleCloudPlatform/ipython-soccer-predictions/blob/master/predict/wc-final.ipynb) [the](https://github.com/facebook/iTorch) [place](http://nbviewer.ipython.org/url/norvig.com/ipython/TSPv3.ipynb) across [many varied languages](https://github.com/ipython/ipython/wiki/IPython-kernels-for-other-languages). It gets used by [data scientists](http://nbviewer.ipython.org/gist/wrobstory/1eb8cb704a52d18b9ee8/Up%20and%20Down%20PyData%202014.ipynb), researchers, analysts, developers, and people in between.
```
Comment (rgbkrk) [inline src/site_source/_posts/2015-01-20-how-did-we-serve-more-than-20000-ipython-notebooks-for-nature.md]: If I'm going to provide links to some users' notebooks, I might as well provide for all of them (that was the intention here and I wanted to make sure I noted it).

**Comment by rgbkrk** [comment]
After review with @danabauer, I'm going to split this up into two articles:

* Meaty bits about lessons learned (likely on personal blog)
* Self contained post about the Nature demo

**Comment by rgbkrk** [comment]
That being said, I'll close this one and break this up.

**Pull request closed without merging**